### ISSUE TITLE
peerDependencies aren't bumped

### ISSUE THREAD

**Issue by StarpTech** [ghiscoding/lerna-lite#333]
### Describe the bug

Hi, I have two packages `A`,`B`. B has a peerDependency to `A`.

```json
{
  "name": "B",
  "peerDependencies": {
    "A": "workspace:^0.105.1"
  },
  "dependencies": {
    "A": "workspace:^0.105.1"
   }
}
```

After publishing `A` to `0.106.0` the version in `dependencies` was updated correctly but not for the peer-dependency.
I tried it as fixed constraints `0.105.0` and without the workspace protocol. In all cases, the peer-dependency was not bumped.

See https://github.com/lerna/lerna/issues/955

### Expectation

I'd expect that the peer dependency is updated to the next compatible version based on the version constraint.

A bumps with semver `^0.105.0` to `0.105.2` => B  bumps with semver `^0.105.0` => `0.105.2` :heavy_check_mark:
A bumps with semver `0.105.0` to `0.106.0` => B bumps with semver `0.105.0` => `0.106.0` :x: because it is not compatible

### Reproduction

See above.

### Lerna config and logs

<details><summary>lerna.json</summary><p>

```json
<!-- Please paste your `lerna.json` here -->
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```
</p></details>

### Environment Info

```shell
| Executable        | Version |
| ----------------- | ------- |
| `lerna --version` | VERSION |
| `npm --version`   | VERSION |
| `yarn --version`  | VERSION |
| `node --version`  | VERSION |

---
OR simply run `npx lerna info` command and copy+paste the result here
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/lerna-lite/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://github.com/ghiscoding/lerna-lite#readme).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/lerna-lite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/lerna-lite/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
If we're reading a few of the comments associated to the issue you referenced, it seems that the original Lerna maintainer (evocateur) purposely left it this way because it's a rather sensitive subject that can be interpreted differently. Basically this [post ](https://github.com/lerna/lerna/issues/955#issuecomment-332074663) is explaining it broadly

> peer deps should never be forcefully updated; updating a minimum peer dep range is a breaking change. Peer dep ranges should always remain as broad as is possible.

and then if we scroll up a little and we click on the issue right above that post which is this [issue](https://github.com/lerna/lerna/issues/1018) 
> peerDependencies shouldn't be updated by `lerna publish`

and start reading what everyone says, it seems that everyone have different opinion on the subject. 

In summary, I'm not so sure that I should deviate from the same implementation as to what Lerna already does, at this point in time, the code in Lerna-Lite follows the same logic as Lerna (which is to say again that evocateur didn't want to change/fix the issue you referenced). I think the best I could do would be to implement 1 of these 2 things
1. update only if it doesn't detect a version range that includes an operator (ie: `"a": ">=1.0.0"`)
2. provide a flag to force update peer dependencies (not sure how to call it though, please provide name suggestion)
   - to be clear the force push is only for when the bump is actually possible
3. we could also go for a mix of 1 and 2, that is when the flag is enabled and we have an version range

I would rather go with option 2 which is to add a flag (need a name) and that wouldn't be too hard to implement in Lerna-Lite because I had already extracted the deps update into a new separate function `retrievePackageDependencies` to fix another peer deps issue you had opened a few months back, so I basically just need to provide `true` as the 2nd argument to this new function and that might just be enough but I don't want to implement it for everyone by default so we would have to go with 1 of these 2 approach

https://github.com/ghiscoding/lerna-lite/blob/9815d64388e14e3eb3eb1970c3a083c0dcb54ac2/packages/core/src/package.ts#L361-L384

**Comment by StarpTech** [comment]
Thank you for the explanation. I'll come up with a proposal.

**Comment by ghiscoding** [comment]
I tested a few code change and found out that there's actually another indirect issue in both Lerna/Lerna-Lite. The way that the code is implemented is that it will only update the first dependency it finds (following this order: 1. `dependencies`, 2. `localDependencies`, 3. `devDependencies` and 4. `peerDependencies`) and so if we define the same dep in both properties `dependencies` and `peerDependencies`, well it will only update the first one and never the peer one. So I think adding an extra loop just for the peer deps might be required to fix this

in other words, if we take the demo you have
```json
{
  "name": "B",
  "peerDependencies": {
    "A": "workspace:^0.105.1"
  },
  "dependencies": {
    "A": "workspace:^0.105.1"
   }
}
```
it would only update the first `dependencies` because of it's primary order even though defining both of these properties should be totally fine (it's only fine for peer deps, technically speaking it's the only time we should allow the same dep repeated)

**Comment by ghiscoding** [comment]
@StarpTech could you please review the PR #362 which will close this issue

**Comment by StarpTech** [comment]
Yes, thank you :+1:
